src/domain/entities/figma-design-identity.test.ts

```diff
@@ -0,0 +1,139 @@
+import {
+	MOCK_FIGMA_FILE_IDENTITY,
+	MOCK_FIGMA_FILE_KEY,
+	MOCK_FIGMA_NODE_ID,
+	MOCK_FIGMA_NODE_IDENTITY,
+} from './testing';
+
+describe('FigmaDesignIdentity', () => {
+	describe('constructor', () => {
+		it('should throw when fileKey is empty string', () => {
+			expect(() => new FigmaDesignIdentity('')).toThrow();
+		});
+
+		it('should throw when nodeId is empty string', () => {
+			expect(() => new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, '')).toThrow();
+		});
+	});
+
+	describe('fromFigmaDesignUrl', () => {
+		it('should return identity when Atlassian design ID does not contain node part', () => {
+			const result = FigmaDesignIdentity.fromAtlassianDesignId(
+				`${MOCK_FIGMA_FILE_KEY}/${MOCK_FIGMA_NODE_ID}`,
+			);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, MOCK_FIGMA_NODE_ID),
+			);
+		});
+
+		it('should return identity when Atlassian design ID contains node part', () => {
+			const result = FigmaDesignIdentity.fromAtlassianDesignId(
+				`${MOCK_FIGMA_FILE_KEY}/${MOCK_FIGMA_NODE_ID}`,
+			);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, MOCK_FIGMA_NODE_ID),
+			);
+		});
+
+		it('should throw when  Atlassian design ID has unexpected format', () => {
+			expect(() => FigmaDesignIdentity.fromFigmaDesignUrl(``)).toThrow();
+		});
+	});
+
+	describe('fromFigmaDesignUrl', () => {
+		it('should return identity when URL does not contain node_id', () => {
+			const designUrl = new URL(
+				`https://www.figma.com/file/${MOCK_FIGMA_FILE_KEY}`,
+			).toString();
+
+			const result = FigmaDesignIdentity.fromFigmaDesignUrl(designUrl);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY),
+			);
+		});
+
+		it('should return identity when URL contains node_id with ":" separator', () => {
+			const designUrl = new URL(
+				`https://www.figma.com/file/${MOCK_FIGMA_FILE_KEY}?node-id=42:1`,
+			).toString();
+
+			const result = FigmaDesignIdentity.fromFigmaDesignUrl(designUrl);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, '42:1'),
+			);
+		});
+
+		it('should return identity when URL contains node_id with encoded ":" separator', () => {
+			const designUrl = new URL(
+				`https://www.figma.com/file/${MOCK_FIGMA_FILE_KEY}?node-id=42%3A1`,
+			).toString();
+
+			const result = FigmaDesignIdentity.fromFigmaDesignUrl(designUrl);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, '42:1'),
+			);
+		});
+
+		it('should return identity when URL contains node_id with "-" separator', () => {
+			const designUrl = new URL(
+				`https://www.figma.com/file/${MOCK_FIGMA_FILE_KEY}?node-id=42-1`,
+			).toString();
+
+			const result = FigmaDesignIdentity.fromFigmaDesignUrl(designUrl);
+
+			expect(result).toStrictEqual(
+				new FigmaDesignIdentity(MOCK_FIGMA_FILE_KEY, '42:1'),
+			);
+		});
+
+		it('should throw when URL has an unexpected format', () => {
```

Comment:
/nit You could structure this as a parameterised test with Jest's [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout)

```typescript
		it.each([
			`https://www.figma.com`,
			`https://www.figma.com/file`,
			`https://www.figma.com?param=file%2Fsome-id`,
		])('should throw when URL has an unexpected format', (url: string) => {
			expect(() => FigmaDesignIdentity.fromFigmaDesignUrl(url)).toThrow();
		});
```

Reply:
Great suggestion! Updated the test accordingly.